File: ios/SOLOREEL/Ads/InterstitialAdGate.swift
import Foundation

/// Tracks episode navigations app-wide so the player can show an admin-uploaded
/// interstitial ad every few episodes instead of on every single one.
enum InterstitialAdGate {
    private static let everyNEpisodes = 3
    private static var count = 0

    /// Call once per new episode load. Returns true when an interstitial should be shown.
    static func shouldShowForNewEpisode() -> Bool {
        count += 1
        return count % everyNEpisodes == 0
    }
}

File: android/app/src/main/java/com/soloreel/app/ads/InterstitialAdGate.kt
package com.soloreel.app.ads

/**
 * Tracks episode navigations app-wide so the player can show an admin-uploaded
 * interstitial ad every few episodes instead of on every single one.
 */
object InterstitialAdGate {
    private const val EVERY_N_EPISODES = 3
    private var count = 0

    /** Call once per new episode load. Returns true when an interstitial should be shown. */
    fun shouldShowForNewEpisode(): Boolean {
        count++
        return count % EVERY_N_EPISODES == 0
    }
}
